### PR TITLE
Require that cipher suites define a record number encryption algorithm

### PR DESCRIPTION
**Pull request by ekr**

### CODE REVIEW

**Review by martinthomson — APPROVED** [review]

File: draft-ietf-tls-dtls13.md [hunk 1]
```diff
@@ -2088,6 +2092,9 @@ registry, defined in {{!TLS13}}, for RequestConnectionId (TBD), and
 NewConnectionId (TBD), as defined in this document.  The value for the
 "DTLS-OK" columns are "Y".
 
+IANA [SHALL mark/has marked] the cipher suites TLS_SHA256_SHA256 and
+TLS_SHA384_SHA_384 DTLS=N.
```
Comment (chris-wood) [inline draft-ietf-tls-dtls13.md:2096]:
```suggestion
IANA [SHALL set] the "DTLS-OK" column value for cipher suites TLS_SHA256_SHA256 and
TLS_SHA384_SHA_384 to "N".
```

**Review by chris-wood — APPROVED** [review]
This one requires WG consensus, I think.

File: draft-ietf-tls-dtls13.md [hunk 2]
```diff
@@ -582,6 +582,10 @@ tag and need no padding. However, some algorithms such as
 TLS_AES_128_CCM_8_SHA256 have a shorter authentication tag and may require padding
 for short inputs.
 
+Future cipher suites which are not based on AES or ChaCha20 MUST define
+their own record sequence number encryption in order to be used with
+DTLS.
```
Comment (kaduk) [inline draft-ietf-tls-dtls13.md:587]: I can't decide whether this requirement merits adding this document as a reference for the ciphersuites registry.

Comment (martinthomson) [reply draft-ietf-tls-dtls13.md:587]: I would say yes, provided that it is additive.

Comment (chris-wood) [reply draft-ietf-tls-dtls13.md:587]: Agreed.

Comment (hannestschofenig) [reply draft-ietf-tls-dtls13.md:587]: What does this imply for the document itself? Do we need to capture it in the IANA consideration section or would we ask IANA to update the references for the TLS Cipher Suites registry at https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml?

File: draft-ietf-tls-dtls13.md (hunk 1 above)
Comment (hannestschofenig) [reply draft-ietf-tls-dtls13.md:2096]: Why do we need to add this IANA entry in this document? Shouldn't https://tools.ietf.org/html/draft-camwinget-tls-ts13-macciphersuites-08 add it to IANA given that they are defining the ciphersuite?

File: draft-ietf-tls-dtls13.md (hunk 2 above)
Comment (kaduk) [reply draft-ietf-tls-dtls13.md:587]: Typically we would have this document's IANA Considerations section request IANA to update the references for the registry to add this document as a reference.

File: draft-ietf-tls-dtls13.md (hunk 1 above)
Comment (kaduk) [reply draft-ietf-tls-dtls13.md:2096]: I agree with Hannes; I don't think we need to do anything about the MAC-only ciphersuites entries.
I've already been in contact with the authors (and DEs) and the authors plan to add text to handle this case.  (Though, they claim that the record numbers will just not be encrypted and that's okay.)

Comment (ekr) [reply draft-ietf-tls-dtls13.md:2096]: They are already in the registry and marked DTLS-OK.

File: draft-ietf-tls-dtls13.md (hunk 2 above)
Comment (ekr) [reply draft-ietf-tls-dtls13.md:587]: Done.

File: draft-ietf-tls-dtls13.md (hunk 1 above)
Comment (kaduk) [reply draft-ietf-tls-dtls13.md:2097]: Yes, they are.  Is that wrong?  Is that more wrong than having MAC-only ciphersuites in the first place?

**Comment by ekr** [comment]
I guess not. Ugh.

File: draft-ietf-tls-dtls13.md [hunk 3]
```diff
@@ -2081,13 +2085,17 @@ IANA is requested to allocate a new value in the "TLS ContentType"
 registry for the ACK message, defined in {{ack-msg}}, with content type 26.
 The value for the "DTLS-OK" column is "Y".  IANA is requested to reserve
 the content type range 32-63 so that content types in this range are not
-allocated.
+allocated.
```
Comment (chris-wood) [inline draft-ietf-tls-dtls13.md:2088]:
```suggestion
allocated.
```

File: draft-ietf-tls-dtls13.md (hunk 1 above)
Comment (ekr) [reply draft-ietf-tls-dtls13.md]: Yeah, I guess we have to live with this.

File: draft-ietf-tls-dtls13.md [hunk 4]
```diff
@@ -2194,6 +2198,10 @@ registry, defined in {{!TLS13}}, for RequestConnectionId (TBD), and
 NewConnectionId (TBD), as defined in this document.  The value for the
 "DTLS-OK" columns are "Y".
 
+IANA is requested to add this RFC as a reference to the TLS Cipher Suite Registry.
+IANA [SHALL set] the "DTLS-OK" column value for cipher suites TLS_SHA256_SHA256 and
+TLS_SHA384_SHA_384 to "N".
+
```
Comment (chris-wood) [inline draft-ietf-tls-dtls13.md:2204]:
```suggestion
```

**Pull request merged**